trin-core/src/jsonrpc/types.rs

```diff
@@ -101,14 +104,56 @@ impl TryFrom<&Value> for PingParams {
     fn try_from(param: &Value) -> Result<Self, Self::Error> {
         match param.as_str() {
             Some(val) => match Enr::from_str(val) {
-                Ok(val) => Ok(PingParams { enr: val }),
+                Ok(enr) => Ok(Self { enr }),
                 Err(_) => Err(ValidationError::new("Invalid enr param")),
             },
             None => Err(ValidationError::new("Missing or empty enr param")),
         }
     }
 }
 
+pub struct FindContentParams {
+    pub enr: Enr,
+    pub content_key: ByteList,
+}
+
+impl TryFrom<Params> for FindContentParams {
+    type Error = ValidationError;
+
+    fn try_from(params: Params) -> Result<Self, Self::Error> {
+        match params {
+            Params::Array(val) => match val.len() {
+                2 => FindContentParams::try_from([&val[0], &val[1]]),
+                _ => Err(ValidationError::new("Expected 2 params")),
+            },
+            _ => Err(ValidationError::new("Expected array of params")),
+        }
+    }
+}
+
+impl TryFrom<[&Value; 2]> for FindContentParams {
+    type Error = ValidationError;
+
+    fn try_from(params: [&Value; 2]) -> Result<Self, Self::Error> {
+        let enr = params[0]
+            .as_str()
+            .ok_or_else(|| ValidationError::new("Empty enr param"))?;
+        let content_key = params[1]
+            .as_str()
+            .ok_or_else(|| ValidationError::new("Empty content key param"))?;
+
+        let enr = match Enr::from_str(enr) {
+            Ok(enr) => enr,
+            Err(_msg) => return Err(ValidationError::new("Invalid enr param")),
+        };
+        let content_key = match decode(content_key) {
```

Comment:
minor nit: I think it's idiomatic in Rust to preserve the namespace for methods, so that we would have `hex::decode` here. This makes it clear that we are decoding hex, and not some other encoding.